### DISCUSSION TITLE
Adapting nvim-lsp-installer to use nvim's new vim.lsp.start

### DISCUSSION THREAD

**Discussion by akinsho** [williamboman/nvim-lsp-installer#745] [Q&A] [answered]
It seems that https://github.com/neovim/neovim/pull/18631 is going to be merged soon and as such I was wondering if there might be a simple mechanism in this plugin to use the installed servers from this plugin easily with this new function.

If you haven't seen the PR the new api looks something like
```lua
vim.lsp.start({
  cmd = {"<path>"},
  name = "server-name",
  root_dir = ...
})
```
A user would then call this in an autocommand or ftplugin file. The main integration point I think being just providing the cmd which is I presume what is already done automagically for nvim-lspconfig.

I don't imagine that people will immediately switch to this api despite being quite straighforward since it requires having to fill out all the server configurations yourself, i.e. root dir and any other default configs, but perhaps lspconfig will just provide the settings in the future so a nice/easy way to add the details from lsp-installer would be good.

I think there is already a way to fetch the path from lsp installer for a server but IIRC it's a bit involved and not advertised up front.

Decided this was more of a discussion than an issue, not sure if that's quite right 🤷🏿‍♂️

**Comment by williamboman** [comment, marked as answer]
Hey! I forgot to reply here 🙈. Thanks for the heads-up, I had not seen that PR before. So I have actually for a long time been exploring symlinking executables to a single location, allowing nvim-lsp-installer to just amend the entire session's PATH once and be done with it.

This would make the entire thing much more seamless without any need to register setup hooks and whatnot (it would also play much more nicely with potential installation of other tools such as linters & formatters to be used with formatter.nvim, null-ls, etc.). Haven't made it happen yet primarily because of Windows support & time.

edit: also some more [context in this comment](https://github.com/Pocco81/dap-buddy.nvim/issues/71#issuecomment-1106527425)

**Reply by akinsho** [reply]
Sounds good, tbh that API whilst nice to see things moving into core and simplified I think isn't something that myself or most people will likely move too straight away since if you work with a lot of languages, lspconfig is still much simpler to use/the default config stuff is essential, so you have a good amount of time anyway

**Reply by williamboman** [reply]
Cross-posting an update here: https://github.com/williamboman/nvim-lsp-installer/discussions/770#discussioncomment-3073074

**Reply by williamboman** [reply]
And another one, this is now available for general use (although still considered alpha stage): https://github.com/williamboman/nvim-lsp-installer/issues/95#issuecomment-1176925455

**Reply by akinsho** [reply]
Just gave it a try and seems to work quite nicely 👍🏿 , I'm not sure how I would get the path to any of the installed packages though to reference elsewhere e.g. in the cmd of lsp config etc.

**Reply by williamboman** [reply]
>  I'm not sure how I would get the path to any of the installed packages though to reference elsewhere e.g. in the cmd of lsp config etc.

I don't think that'll be needed. Executables installed via mason.nvim are available in your `PATH` automatically (after installing, try `:echo executable("lua-language-server")` or `:echo exepath("lua-language-server")` for example - or even better launch a toggleterm instance and check `type -a lua-language-server` etc. 🙂)
